### ISSUE TITLE
Modal close prevents aria-live text from being read

### ISSUE THREAD

**Issue by carc1n0gen** [twbs/bootstrap#28727]
- macOS Majave
- Safari

**Problem**

When closing a modal, some animations are done, and then the button that opened the modal is focused and read again.  After the close I wanted to add some aria-live text to be read out when ajax request is complete. For some reason even though I have aria-live="assertive" it isn't read

**Weird workaround**

If I wrap the append to the aria-live-box in a 1000ms timeout, it starts to read the button, but then is interrupted by the aria-live text.

**Other thoughts**

I believe this has something specific to do with the modal because when I just use an onclick for a plain button on the page and append the aria-live text it is read no problem.

Maybe someone has experienced this and found another workaround with a better experience than the voice interrupting itself.

```js
$.post('/admin/users/create', formdata)
        .done(function(data) {
            $('tbody').append(formatTemplate(template, {
                userid: data.id,
                username: data.username,
                email: data.email
            }));
            $('#new-user-modal').modal('hide');
            $('#new-user-form').trigger('reset');
            $('#aria-live-box').append(
                '<p>Created user ' + data.username + ' successfully.</p>'
            );
        })
```

**Comment by patrickhlauke** [comment]
Per the issue template, we really need a live reduced test case example of the issue.

**Comment by carc1n0gen** [comment]
Issue:

https://jsfiddle.net/p0va9hex/4/

With the weird workaround:

https://jsfiddle.net/p0va9hex/3/

In the first sample, the aria-live text is never read.  Something to do with focus to the button that opened the modal I think.

In the second sample, I delay adding the aria-live text by 1000ms, the button is still focussed and it starts to read it, and then the aria-live interrupts and is read.

I'd like to find a way that I don't have the voice over interrupt itself.  It's a bit of a poor experience

**Comment by patrickhlauke** [comment]
FWIW, the first example works fine with Chrome/NVDA/Windows (not had time to test other combinations). Live regions are fiddly and have quirks in different browsers/assistive technologies, so sometimes they do require extra fixes and workarounds to work correctly. Safari is notorious for being a bit...particular about when it announces things.
To avoid interruption, you may want to experiment with a longer delay and using `aria-live="polite"` instead.
But generally, this is not an issue with Bootstrap itself, as it behaves correctly here (setting the focus back to the control that opened the modal), but more of a "how to" that goes beyond core functionality. Suggest further discussion should be done on Stackoverflow or similar - see https://github.com/twbs/bootstrap/blob/master/.github/CONTRIBUTING.md#using-the-issue-tracker

**Comment by carc1n0gen** [comment]
I'm on macOS Mojave, so I'm using the built in macOS voice over. And using that, I see the same misbehaviour regardless of browser. In chrome, Firefox, and safari, I see exactly the same behaviour I previously described.

I didn't mean to turn this in to a how to, I would like this to be treated a bug. It's an incompatibility with the macOS voice over.

**Comment by patrickhlauke** [comment]
No it is not. It is a bug/behaviour of VoiceOver in general then, not Safari/VO specifically. As said, the behaviour from Bootstrap's side (setting focus onto the control that originally opened the modal) is the correct one. Your desire to now update a live region immediately afterwards, and timing issues that then come from that in VO, are outside of Bootstrap's problems and relate directly to working around timing issues with VO on your part (and as said, `aria-live="polite"` may be part of the solution for your specific situation).

**Comment by carc1n0gen** [comment]
aria-live="polite" and a short timeout seemed to be the solution.  Sorry for not listening to that sooner.
